crates/title_bar/src/title_bar.rs

```diff
@@ -531,26 +539,34 @@ impl TitleBar {
                 })
         }?;
 
-        Some(
-            Button::new("project_branch_trigger", branch_name)
-                .color(Color::Muted)
-                .style(ButtonStyle::Subtle)
-                .label_size(LabelSize::Small)
-                .tooltip(move |window, cx| {
-                    Tooltip::with_meta(
-                        "Recent Branches",
-                        Some(&zed_actions::git::Branch),
-                        "Local branches only",
-                        window,
-                        cx,
-                    )
-                })
-                .on_click(move |_, window, cx| {
-                    let _ = workspace.update(cx, |_this, cx| {
-                        window.dispatch_action(zed_actions::git::Branch.boxed_clone(), cx);
-                    });
-                }),
-        )
+        let mut branch_button = Button::new("project_branch_trigger", branch_name)
```

Comment:
This works, but we can reduce the diff and stick to the builder-like API with `.when(TitleBarSettings::get_global(cx).show_branch_icon, |branch_button| ...)`.

Let's try doing that?

Reply:
Done !